pkg/lib/packagekit.js

```diff
@@ -302,30 +302,39 @@ const validSubscriptionStates = [0, 3, 4];
 
 /**
  * Check Red Hat subscription-manager if if this is an unregistered RHEL
- * system. If subscription-manager is not installed, nothing happens.
+ * system. If subscription-manager is not installed or enabled in yum/dnf,
+ * nothing happens.
  *
  * callback: Called with a boolean (true: registered, false: not registered)
  *           after querying subscription-manager once, and whenever the value
  *           changes.
  */
 export function watchRedHatSubscription(callback) {
-    // check if this is an unregistered RHEL system; if subscription-manager is not installed, ignore
-    var sm = cockpit.dbus("com.redhat.SubscriptionManager");
-    sm.subscribe(
-        { path: "/EntitlementStatus",
-          interface: "com.redhat.SubscriptionManager.EntitlementStatus",
-          member: "entitlement_status_changed"
-        },
-        (path, iface, signal, args) => callback(validSubscriptionStates.indexOf(args[0]) >= 0)
-    );
-    sm.call(
-        "/EntitlementStatus", "com.redhat.SubscriptionManager.EntitlementStatus", "check_status")
-            .done(result => callback(validSubscriptionStates.indexOf(result[0]) >= 0))
-            .fail(ex => {
-                if (ex.problem != "not-found")
-                    console.warn("Failed to query RHEL subscription status:", JSON.stringify(ex));
-            }
-            );
+    // first check if subscription-manager is enabled in yum/dnf
+    cockpit.file("/etc/yum/pluginconf.d/subscription-manager.conf").read()
+            .then(contents => {
+                if (!contents || contents.indexOf("enabled=1") < 0)
```

Comment:
wouldn't it be allowed to have `disabled = 1` in `/etc/yum/pluginconf.d/subscription-manager.conf`? (spaces around `=`) Or possibly even `disabled = true`?

Reply:
Really, there is a `disabled=` *and* and `enabled=`? The [docs](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system_administrators_guide/sec-yum_plugins) only talk about `enabled=`.

I did this because that's how the RPM sets up the configuration file on RHEL by default and to keep things simple. The docs don't mention spacing, but I can use some regexp that allows it for robustness.

Reply:
Fixed.